app/helpers/application_filter_helper.rb

```diff
@@ -0,0 +1,248 @@
+  end
+
+  def apply_status_filter(application_forms)
+    application_forms.joins(:application_choices).where(application_choices: { status: applied_filters[:status] })
+  end
+
+  def apply_provider_id_filter(application_forms)
+    application_forms
+      .joins(:application_choices)
+      .where('application_choices.provider_ids @> ?', "{#{applied_filters[:provider_id]}}")
+  end
+
+private
+
+  def year_filter
+    cycle_options = RecruitmentCycle::CYCLES.map do |year, label|
+      {
+        value: year,
+        label: label,
+        checked: applied_filters[:year]&.include?(year),
+      }
+    end
+
+    {
+      type: :checkboxes,
+      heading: 'Recruitment cycle year',
+      name: 'year',
+      options: cycle_options,
+    }
+  end
+
+  def search_filter
+    {
+      type: :search,
+      heading: 'Name, email or reference',
+      value: applied_filters[:q],
+      name: 'q',
+    }
+  end
+
+  def search_by_application_choice_filter
+    {
+      type: :search,
+      css_classes: 'govuk-input--width-5',
+      heading: 'Provider application ID',
+      value: applied_filters[:application_choice_id],
+      name: 'application_choice_id',
+    }
+  end
+
+  def phase_filter
+    {
+      type: :checkboxes,
+      heading: 'Phase',
+      name: 'phase',
+      options: [
+        {
+          value: 'apply_1',
+          label: 'Apply 1',
+          checked: applied_filters[:phase]&.include?('apply_1'),
+        },
+        {
+          value: 'apply_2',
+          label: 'Apply 2',
+          checked: applied_filters[:phase]&.include?('apply_2'),
+        },
+      ],
+    }
+  end
+
+  def interviews_filter
+    {
+      type: :checkboxes,
+      heading: 'Interviews',
+      name: 'interviews',
+      options: [
+        {
+          value: 'has_interviews',
+          label: 'Has interviews',
+          checked: applied_filters[:interviews]&.include?('has_interviews'),
+        },
+      ],
+    }
+  end
+
+  def training_provider_filter
+    providers = Provider
+                .joins(:courses)
+                .where(id: applied_filters['provider_id'])
+                .or(Course.where(accredited_provider_id: applied_filters['provider_id']),)
+                .distinct
+
+    provider_options = providers.map do |provider|
+      {
+        value: provider.id,
+        label: provider.name,
+        checked: applied_filters[:training_provider]&.include?(provider.id.to_s),
+      }
+    end
+
+    {
+      type: :checkboxes,
+      heading: 'Training providers',
+      name: 'training_provider',
+      options: provider_options,
+    }
+  end
+
+  def accredited_provider_filter
+    courses = Course
+              .joins(:provider)
+              .includes([:accredited_provider])
+              .where(provider: { id: applied_filters['provider_id'] })
+              .or(Course.where(accredited_provider_id: applied_filters['provider_id']))
+              .distinct
```

Comment:
I think this query is actually wrong as it should really only be providing accrediting providers where a candidate has actually applied to their course. Ngl my brain is absolutely melted though.

Help me @stevehook .... you're my only hope

Reply:
Sorry I know we discussed this @davidgisbey but now that I've taken another a look it does seem like we would only want to return accredited providers for this filter right?

Reply:
Still trying to understand this properly but it makes sense to limit this set of options to accredited providers that have received applications. Doesn't the same thing apply to `training_provider_filter` above?